src/LinkDotNet.NCronJob/Scheduler/QueueWorker.cs

```diff
@@ -94,6 +113,12 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         }
     }
 
+    private void RescheduleJobs(object? sender, EventArgs e)
+    {
+        queueWaiter.Release();
```

Comment:
Is this really the correct way?
Or:
```csharp
    private void RescheduleJobs(object? sender, EventArgs e)
    {
        rescheduleTrigger.Cancel();
        queueWaiter.Release();
    }
```

Reply:
Cancel is for Jobs that are currently executing. I don't think rescheduling should implicitly cancel a running job. However, I could see from a dashboard/CLI the option to cancel the job if it's currently running and you're trying to reschedule. So perhaps add a parameter called `forceCancel = false` where you cancel (if it's running) if the user chooses to cancel.

Reply:
Just so that we understand each other right - the idea is not to cancel any job.

The code is for the following:
Imagine you have a job that runs hourly. But now you want to trigger an instant job (which with this PR goes through the same pipeline). Obvious our current approach was to wait exactly one hour - execute the job - and grab the next one.

What my code does is basically tell the JobQueue: "Hey, you have a new entry that might be eligible to execute before the current one - therefore, abort the current waiting and retry".

So no job gets canceled. We basically just go back a step and re-evaluate.

For sure I will add appropriate tests for that!

Reply:
Added a specific test to document that runs will not be cancelled.